src/ft2font.cpp

```diff
@@ -207,32 +206,42 @@ FT2Font::get_path(std::vector<double> &vertices, std::vector<unsigned char> &cod
     codes.push_back(CLOSEPOLY);
 }
 
-FT2Font::FT2Font(FT_Open_Args &open_args,
-                 long hinting_factor_,
-                 std::vector<FT2Font *> &fallback_list,
-                 FT2Font::WarnFunc warn, bool warn_if_used)
-    : ft_glyph_warn(warn), warn_if_used(warn_if_used), image({1, 1}), face(nullptr),
+FT2Font::FT2Font(long hinting_factor_, std::vector<FT2Font *> &fallback_list,
+                 bool warn_if_used)
+    : warn_if_used(warn_if_used), image({1, 1}), face(nullptr), fallbacks(fallback_list),
       hinting_factor(hinting_factor_),
       // set default kerning factor to 0, i.e., no kerning manipulation
       kerning_factor(0)
 {
     clear();
+}
+
+FT2Font::~FT2Font()
+{
+    close();
+}
+
+void FT2Font::open(FT_Open_Args &open_args)
+{
     FT_CHECK(FT_Open_Face, _ft2Library, &open_args, 0, &face);
     if (open_args.stream != nullptr) {
         face->face_flags |= FT_FACE_FLAG_EXTERNAL_STREAM;
     }
-    // Set fallbacks
-    std::copy(fallback_list.begin(), fallback_list.end(), std::back_inserter(fallbacks));
 }
 
-FT2Font::~FT2Font()
+void FT2Font::close()
 {
+    // This should be idempotent, in case a user manually calls close before the
```

Comment:
It looks like we will always call close twice in the normal path (first from the subclass and then from here)?

Reply:
I mean it looks like the sub-class destructor calls `close` and then the descructor calls `close` so `close` is always called 2x?

Reply:
Yes, that was mostly for safety's sake. I don't think there would be anything making an `FT2Font` object directly, but just in case we refactor things and it happens, the call is there. Everything should be NULL'd out though so it won't crash.